hyperspy/_signals/lazy.py

```diff
@@ -84,17 +84,46 @@ class LazySignal(BaseSignal):
     """
     _lazy = True
 
-    def compute(self, progressbar=True):
-        """Attempt to store the full signal in memory.."""
+    def compute(self, progressbar=True, close_file=False):
+        """Attempt to store the full signal in memory.
+
+        close_file: bool
+            If True, attemp to close the file associated with the dask
+            array data if any. Note that closing the file will make all other
+            associated lazy signals inoperative.
+
+        """
         if progressbar:
             cm = ProgressBar
         else:
             cm = dummy_context_manager
         with cm():
-            self.data = self.data.compute()
+            da = self.data
+            data = da.compute()
+            if close_file:
+                self.close_file()
+            self.data = data
         self._lazy = False
         self._assign_subclass()
 
+    def close_file(self):
+        """Closes the associated data file if any.
+
+        Currently it only supports closing the file associated with a dask
+        array created from an h5py DataSet (default HyperSpy hdf5 reader).
+
+        """
+        arrkey = None
+        for key in self.data.dask.keys():
+            if "array-original" in key:
+                arrkey = key
+                break
+        if arrkey:
+            try:
+                self.data.dask[arrkey].file.close()
```

Comment:
Yes, that's normal because the COO object is indeed not a file.

Reply:
ok, but the file should be closed somewhere?

Reply:
The file is already closed because the data  is the stream data itself that is not loaded lazily. What is computed lazily here is the conversion between the sparse array and the dense array.

Reply:
From line 543, it looks like the file is not closed:
``` python
finally:
    if not self.lazy:
        f.close()
```
and as it is now, we should not closed it, because it needs to stay open to access the spectrum and the images.
Should the `close_file` take into account this particular case and not add the error to the logger, since it is misleading?

Reply:
You are right, we shouldn't close it because otherwise the lazy spectra and images will get broken. And actually, calling close doesn't actually close the file for the reason above. We could not attempt to close the file when it is a SI in the tests. However, by leaving it as is we are actually testing that the code is ``except`` is working fine. So maybe it's not that bad after all?

A side note, the addition of this feature makes only sense in windows, where, for some reason, the files are not closed after the object is deleted even after running the garbage collector. So, this is a workaround for a bug (or maybe a limitation) elsewhere and most users won't actually need to use it. It is useful here to make the tests pass in appveyor. The alternative is not removing the files as it was previously done.

Reply:
Although I would prefer that this tests has his own in the unit tests of the lazy signal (because it may look confusing to see this error when testing emd file), if you put a comment explaining this, then it is fine.